files/internals/cports.common

```diff
@@ -89,27 +175,57 @@ PROTO="icmp"
 if [ "$ICMP_LIM" == "" ]; then
 	ICMP_LIM=0
 fi
-if [ "$(echo $ICMP_LIM | tr '/' ' ' | awk '{print$1}')" -gt "0" ]; then
-	ICMP_EARGS="-m limit --limit $ICMP_LIM"
+if [ $EXECUTOR == "nft" ]; then
+   if [ "$(echo $ICMP_LIM | tr '/' ' ' | awk '{print$1}')" -gt "0" ]; then
+	ICMP_EARGS="limit rate $ICMP_LIM burst 5 packets"
+   else
+	ICMP_EARGS=""
+   fi
 else
+   if [ "$(echo $ICMP_LIM | tr '/' ' ' | awk '{print$1}')" -gt "0" ]; then
+	ICMP_EARGS="-m limit --limit $ICMP_LIM"
+   else
 	ICMP_EARGS=""
+   fi
 fi
 IG_ICMP_TYPES=`echo $IG_ICMP_TYPES | tr ',' ' '`
 	for i in `echo $IG_ICMP_TYPES`; do
          if [ ! "$i" == "" ]; then
                 i=`echo $i | tr '[:upper:]' '[:lower:]'`
                 if [ "$i" == "all" ]; then
-                  $IPT $IPT_FLAGS -A INPUT -p icmp -d $VNET  -s 0/0 $ICMP_EARGS -j ACCEPT
+	          if [ $EXECUTOR == "nft" ]; then
+                     if [ "$USE_IPV6" == "1" ]; then
+                        $NFT add rule ip6 filter INPUT meta l4proto icmp ip6 daddr $VNET $ICMP_EARGS counter accept
+                     else
+                        #$IPT $IPT_FLAGS -A INPUT -p icmp -d $VNET  -s 0/0 $ICMP_EARGS -j ACCEPT
+                        $NFT add rule ip filter INPUT ip protocol icmp ip daddr $VNET $ICMP_EARGS counter accept
+                     fi
+		  else
+                     $IPT $IPT_FLAGS -A INPUT -p icmp -d $VNET  -s 0/0 $ICMP_EARGS -j ACCEPT
+		  fi
                   eout "{glob} opening inbound $PROTO all on $VNET"
                 else
-                  $IPT $IPT_FLAGS -A INPUT -p icmp --icmp-type $i -d $VNET  -s 0/0 $ICMP_EARGS -j ACCEPT
+	          if [ $EXECUTOR == "nft" ]; then
+	             if [ "$USE_IPV6" == "1" ]; then
+	                 $NFT add rule ip6 filter INPUT ip6 daddr $VNET icmp code $i $ICMP_EARGS counter accept
```

Comment:
should we do a mapping? because currently it mix with number and text

man nft for mapping from digits to text
```

table ip filter {
	chain INPUT {
		ip daddr 172.17.0.3 icmp code port-unreachable counter packets 0 bytes 0 accept
		ip daddr 172.17.0.3 icmp code 5 counter packets 0 bytes 0 accept
		ip daddr 172.17.0.3 icmp code 11 counter packets 0 bytes 0 accept
		ip daddr 172.17.0.3 icmp code net-unreachable counter packets 0 bytes 0 accept
		ip daddr 172.17.0.3 icmp code 30 counter packets 0 bytes 0 accept
		ip daddr 172.17.0.3 icmp code 8 counter packets 0 bytes 0 accept
	}

	chain OUTPUT {
	}
}
table ip6 filter {
	chain INPUT {
		ip6 daddr fd00::242:ac11:3 icmp code port-unreachable counter packets 0 bytes 0 accept
		ip6 daddr fd00::242:ac11:3 icmp code 5 counter packets 0 bytes 0 accept
		ip6 daddr fd00::242:ac11:3 icmp code 11 counter packets 0 bytes 0 accept
		ip6 daddr fd00::242:ac11:3 icmp code net-unreachable counter packets 0 bytes 0 accept
		ip6 daddr fd00::242:ac11:3 icmp code 30 counter packets 0 bytes 0 accept
		ip6 daddr fd00::242:ac11:3 icmp code 8 counter packets 0 bytes 0 accept
	}

	chain OUTPUT {
	}
}

```

Reply:
e1fc8f0